packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js

```diff
@@ -2226,4 +2227,36 @@ export default class Server {
       },
     });
   }
+
+  getOperatingSystemDetailByServerName(serviceName) {
+    const deffered = this.$q.defer();
+    this.$http
+      .get(`${this.path.dedicatedServer}/${serviceName}`)
+      .then(({ os }) => {
+        const osCodes = [
+          OPERATING_SYSTEM_ENUM.NONE,
+          OPERATING_SYSTEM_ENUM.BRING_YOUR_OWN_IMAGE,
+        ];
+        if (!os || osCodes.includes(os)) {
+          deffered.resolve({
+            description:
+              os === OPERATING_SYSTEM_ENUM.BRING_YOUR_OWN_IMAGE
+                ? 'server_configuration_distribution_byoi'
+                : 'server_configuration_distribution_none',
+          });
+        } else {
+          this.getOperatingSystemDetailByOS(os)
+            .then(({ data }) => deffered.resolve(data))
+            .catch((error) => deffered.reject(error));
+        }
+      })
+      .catch((error) => deffered.reject(error));
+    return deffered.promise;
+  }
+
+  getOperatingSystemDetailByOS(templateName) {
+    return this.$http.get(
+      `${this.path.proxy}/dedicated/installationTemplate/${templateName}`,
+    );
+  }
```

Comment:
I'm not sure but i think can simplified, but it will be test:

```suggestion
getOperatingSystemDetailByServerName(serviceName) {
    return this.$http
      .get(`${this.path.dedicatedServer}/${serviceName}`)
      .then(({ os }) => {
        const osCodes = [
          OPERATING_SYSTEM_ENUM.NONE,
          OPERATING_SYSTEM_ENUM.BRING_YOUR_OWN_IMAGE,
        ];
        if (!os || osCodes.includes(os)) {
          return {
            description:
              os === OPERATING_SYSTEM_ENUM.BRING_YOUR_OWN_IMAGE
                ? this.$translate.instant(
                    'server_configuration_distribution_byoi',
                  )
                : this.$translate.instant(
                    'server_configuration_distribution_none',
                  ),
          };
        }
        return {
          description: this.getOperatingSystemDetailByOS(os),
        };
      });
  }

  getOperatingSystemDetailByOS(templateName) {
    return this.$http
      .get(`${this.path.proxy}/dedicated/installationTemplate/${templateName}`)
      .then(({ data }) => data);
  }
```

Reply:
For the translation part, yes you're right but `getOperatingSystemDetailByOS` return a promise so it can't be set into `{description: (return value)}`